be/src/formats/parquet/file_writer.cpp

```diff
@@ -177,139 +192,718 @@ void ParquetBuildHelper::build_compression_type(::parquet::WriterProperties::Bui
     }
 }
 
+std::shared_ptr<::parquet::schema::GroupNode> ParquetBuildHelper::make_schema(
+        const std::vector<std::string>& file_column_names, const std::vector<ExprContext*>& output_expr_ctxs) {
+    ::parquet::schema::NodeVector fields;
+
+    for (int i = 0; i < output_expr_ctxs.size(); i++) {
+        auto column_expr = output_expr_ctxs[i]->root();
+        auto nodePtr = _make_schema_node(
+                file_column_names[i], column_expr->type(),
+                column_expr->is_nullable() ? ::parquet::Repetition::OPTIONAL : ::parquet::Repetition::REQUIRED);
+        DCHECK(nodePtr != nullptr);
```

Comment:
what if it is some type we don't support?

Reply:
Fixed. Report error on unsupported datatype now.